src/App.js

```diff
@@ -1,73 +1,83 @@
-
-  },[])
-  const location =useLocation();
-const editModalDisplay=useSelector(state=>state?.posts?.editModalDisplay);
-const contentEditPost=useSelector(state=>state?.posts?.editedPost);
-  return (
-    <div className="App">
-      <Routes>
-        <Route element={<RestrictedRoute />}>
-          <Route path="/" element={<LandingPage />} />
-        </Route>
-        <Route element={<PrivateRoute />}>
-          <Route path="/homePage" element={<HomePage />} />
-          <Route path={`/profilePage/:profileId`} element={<ProfilePage />} />
-          <Route path={`/settings`} element={<SettingsPage />} />
-          <Route path={`/bookMarkPage`} element={<BookMarkPage />} />
-          <Route path="/post/:postId" element={<PostPage />} />
-        </Route>
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(checkToken());
+    dispatch(getAllUsers());
+    dispatch(getAllPosts());
+     dispatch(getBookMarks());
+  }, []);
+  const location = useLocation();
+  const editModalDisplay = useSelector(
+    (state) => state?.posts?.editModalDisplay
+  );
+  const contentEditPost = useSelector((state) => state?.posts?.editedPost);
+return (
+  <div className="App">
+    <Routes>
+      <Route element={<RestrictedRoute />}>
+        <Route path="/" element={<LandingPage />} />
+      </Route>
+      <Route element={<PrivateRoute />}>
+        <Route path="/homePage" element={<HomePage />} />
+        <Route path={`/profilePage/:profileId`} element={<ProfilePage />} />
+        <Route path={`/settings`} element={<SettingsPage />} />
+        <Route path={`/bookMarkPage`} element={<BookMarkPage />} />
+        <Route path="/post/:postId" element={<PostPage />} />
+        <Route path="/explore" element={<ExplorePage />} />
+      </Route>
 
-        <Route path="*" element={<Page404 />} />
-        <Route path="/mock" element={<Mockman />} />
-      </Routes>
+      <Route path="*" element={<Page404 />} />
+      <Route path="/mock" element={<Mockman />} />
+    </Routes>
 
-      {editModalDisplay && (
-        <PostEditModal
-          textArea={
-            <textarea
-              role="textbox"
-              type="text"
-              name="post"
-              className="postInput__content-input"
-              placeholder="Write something in this"
-              rows="3"
-              onChange={(e) => {
-                dispatch(editPostContent(e.target.value));
-              }}
-              value={contentEditPost}
-            ></textarea>
-          }
-        ></PostEditModal>
-      )}
-      <ToastContainer />
-    </div>
-  );
+    {editModalDisplay && (
+      <PostEditModal
+        textArea={
+          <textarea
+            role="textbox"
+            type="text"
+            name="post"
+            className="postInput__content-input"
+            placeholder="Write something in this"
+            rows="3"
+            onChange={(e) => {
+              dispatch(editPostContent(e.target.value));
```

Comment:
I think calling reducer in every change is not a good approach

Reply:
ohh this is just a global state use to keep the content globally because otherwise there would be a lot prop drilling which I wanted to avoid. That's why its this way